teamcity/tests/test_teamcity.py

```diff
@@ -137,3 +120,100 @@ def test_config(test_case, extra_config, expected_http_kwargs):
         http_wargs.update(expected_http_kwargs)
 
         r.assert_called_with(ANY, **http_wargs)
+
+
+@pytest.mark.parametrize(
+    'build_config, expected_error',
+    [
+        pytest.param(
+            {'projects': {'project_id': {}}}, 'Failed to establish a new connection', id="One `projects` config"
+        ),
+        pytest.param(
+            {'build_configuration': 'build_config_id'},
+            'Failed to establish a new connection',
+            id="One `build_configurations` config",
+        ),
+        pytest.param(
+            {'projects': {'project_id': {}}, 'build_configuration': 'build_config_id'},
+            'Only one of `projects` or `build_configuration` may be configured, not both.',
+            id="Redundant configs",
+        ),
+    ],
+)
+def test_validate_config(dd_run_check, build_config, expected_error, caplog):
+    """
+    Test that the `build_configuration` config options are properly configured in Python 2 prior to running check.
+    Note: The properly configured test cases would be expected to have a `Failed to establish a new connection`
+    exception.
+    """
+    caplog.clear()
+    config = {'server': 'server.name', 'use_openmetrics': False}
+
+    instance = deepcopy(config)
+    instance.update(build_config)
+
+    check = TeamCityRest('teamcity', {}, [instance])
+
+    if PY2 and instance.get('projects'):
+        expected_error = (
+            '`projects` option is not supported for Python 2. '
+            'Use the `build_configuration` option or upgrade to Python 3.'
+        )
+
+    with pytest.raises(Exception, match=expected_error):
+        dd_run_check(check)
+
+
+def test_collect_build_stats(aggregator, mock_http_response, rest_instance, teamcity_rest_check):
+    check = teamcity_rest_check(rest_instance)
+    check.build_tags = BUILD_TAGS
+
+    mock_http_response(file_path=get_fixture_path("build_stats.json"))
+    check._collect_build_stats(NEW_SUCCESSFUL_BUILD)
```

Comment:
I'm a bit reluctant about these tests that rely on calling the internal function `_collect_build_stats()`. This would mean that the tests would break if any internal implementation details change. Would it be possible to still have the same test although only using `check()`? If it would be too much work, then I think this is fine.